server/public/plugin/supervisor.go

```diff
@@ -54,48 +96,31 @@ func newSupervisor(pluginInfo *model.BundleInfo, apiImpl API, driver Driver, par
 		},
 	}
 
-	executable := pluginInfo.Manifest.GetExecutableForRuntime(runtime.GOOS, runtime.GOARCH)
-	if executable == "" {
-		return nil, fmt.Errorf("backend executable not found for environment: %s/%s", runtime.GOOS, runtime.GOARCH)
-	}
-
-	executable = filepath.Clean(filepath.Join(".", executable))
-	if strings.HasPrefix(executable, "..") {
-		return nil, fmt.Errorf("invalid backend executable: %s", executable)
-	}
-
-	executable = filepath.Join(pluginInfo.Path, executable)
-
-	cmd := exec.Command(executable)
-
-	// This doesn't add more security than before
-	// but removes the SecureConfig is nil warning.
-	// https://mattermost.atlassian.net/browse/MM-49167
-	pluginChecksum, err := getPluginExecutableChecksum(executable)
-	if err != nil {
-		return nil, errors.Wrapf(err, "unable to generate plugin checksum")
-	}
-
-	sup.client = plugin.NewClient(&plugin.ClientConfig{
+	clientConfig := &plugin.ClientConfig{
 		HandshakeConfig: handshake,
 		Plugins:         pluginMap,
-		Cmd:             cmd,
 		SyncStdout:      wrappedLogger.With(mlog.String("source", "plugin_stdout")).StdLogWriter(),
 		SyncStderr:      wrappedLogger.With(mlog.String("source", "plugin_stderr")).StdLogWriter(),
 		Logger:          hclogAdaptedLogger,
 		StartTimeout:    time.Second * 3,
-		SecureConfig: &plugin.SecureConfig{
-			Checksum: pluginChecksum,
-			Hash:     sha256.New(),
-		},
-	})
+	}
+	for _, opt := range opts {
+		err := opt(clientConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to apply option")
+		}
+	}
+
+	sup.client = plugin.NewClient(clientConfig)
 
 	rpcClient, err := sup.client.Client()
 	if err != nil {
 		return nil, err
 	}
 
-	sup.pid = cmd.Process.Pid
+	if clientConfig.Cmd != nil {
+		sup.pid = clientConfig.Cmd.Process.Pid
+	}
```

Comment:
Turns out the `pid` isn't required, but I left this intact for now in the normal plugin execution case.

Reply:
Yes this is leftover from the original implementation of the plugin health check job. We removed the explicit check for "is that process still running", as there were issues running that on Windows.